StripeApplePay/StripeApplePay/Source/ApplePayContext/STPApplePayContext.swift

```diff
@@ -536,121 +557,73 @@ public class STPApplePayContext: NSObject, PKPaymentAuthorizationControllerDeleg
-                        DispatchQueue.main.async {
-                            self.callDidCompleteDelegate(status: .success, error: nil)
-                            self._end()
-                        }
+            } else {
+                completion(PKPaymentAuthorizationStatus.failure, error)
+            }
+        }
+        // Helper 2: Handle success
+        func handleSuccess() {
+            self.paymentState = .success
+            if self.didCancelOrTimeoutWhilePending {
+                self.authorizationController.dismiss {
+                    DispatchQueue.main.async {
+                        self.callDidCompleteDelegate(status: .success, error: nil)
+                        self._end()
                     }
-                } else {
-                    completion(PKPaymentAuthorizationStatus.success, nil)
                 }
-                return
-            case .pending, .notStarted:
-                let errorAnalytic = ErrorAnalytic(event: .unexpectedApplePayError,
-                                                  error: Error.invalidFinalState)
-                self.analyticsClient.log(analytic: errorAnalytic)
-                stpAssertionFailure("Invalid final state")
-                return
+            } else {
+                completion(PKPaymentAuthorizationStatus.success, nil)
             }
         }
 
         // 1. Create PaymentMethod
         StripeAPI.PaymentMethod.create(apiClient: apiClient, payment: payment, clientAttributionMetadata: clientAttributionMetadata) { result in
-            guard let paymentMethod = try? result.get(), self.authorizationController != nil else {
-                if case .failure(let error) = result {
-                    let errorMessage = "Failed on token creation"
-                    let errorAnalytic = ErrorAnalytic(event: .unexpectedApplePayError,
-                                                      error: error,
-                                                      additionalNonPIIParams: [
-                                                        "error_message": errorMessage
-                                                      ])
-                    handleFinalState(.error, error, errorAnalytic)
-                } else {
-                    let errorMessage = "Failed while creating paymentMethod due to internal error"
-                    let errorAnalytic = ErrorAnalytic(event: .unexpectedApplePayError,
-                                                      error: InternalError.invalidState,
-                                                      additionalNonPIIParams: [
-                                                        "error_message": errorMessage
-                                                      ])
-
-                    self.analyticsClient.log(analytic: errorAnalytic, apiClient: self.apiClient)
-                    handleFinalState(.error, nil, errorAnalytic)
-                }
+            guard !self.didFinish else {
```

Comment:
Replaced the old `guard self.authorizationController != nil` behavior with explicit `guard !self.didFinish`